src/main.rs

```diff
@@ -0,0 +1,102 @@
+use anyhow::Error;
+use clap::Parser;
+use dns::{DnsRecord, DnsRecordCreate};
+use reqwest::Url;
+
+use crate::{config::Config, consul::ConsulClient, nomad::NomadDnsTag};
+
+mod config;
+mod consul;
+mod dns;
+mod nomad;
+
+#[tokio::main]
+async fn main() {
+    let config = Config::parse();
+    print!("Config: {:?}", config);
+
+    // Initialize Consul Client
+    let consul_client = ConsulClient::new(
+        Url::parse(&config.consul_address).expect("Invalid URL"),
+        config.consul_datacenter.clone(),
+    )
+    .expect("Failed to create Consul client");
+
```

Comment:
You need a loop here somewhere (with a sleep) otherwise this will only run once.